client/src/pages/SearchMovie/AddMovieButton.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react'
+
+import { Ticket, LeftCutout, RightCutout } from '../../styles/Ticket'
+import { AddButton } from './styled'
+import { authFetch } from '../../auth'
+import { useConfiguration } from '../../providers/Configuration'
+import { useMovies } from '../../providers/Movies'
+
+function findMovieId(movies, movieOmdbId) {
+  for (const movie of movies) {
+    if (movie.omdb_id === movieOmdbId) {
+      return movie.id
+    }
+  }
+}
+
+const AddMovieButton = ({ movie }) => {
+  const { apiUrl } = useConfiguration()
+  const { movies, getUserSavedMovies } = useMovies()
+
+  async function addMovie(movie) {
+    const body = {
+      poster_url: movie.Poster,
+      name: movie.Title,
+      omdb_id: movie.imdbID,
+    }
+    try {
+      const response = await authFetch(`${apiUrl}/watchlist`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(body),
+      })
+      const data = await response.json()
+      console.log(data)
+      getUserSavedMovies()
+      movie.isAdded = true
```

Comment:
Ideally, best practice is to avoid mutating variables whenever possible. When you mutate values, it can quickly become tough to follow the control flow in your app logic. 

But since we discussed the tradeoffs already, this is ok.